hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -780,20 +787,40 @@ def calculate_net_pay(self):
 				relieving_date=self.relieving_date,
 			)[1]
 
-		self.gross_pay = self.get_component_totals("earnings", depends_on_payment_days=1)
-		self.base_gross_pay = flt(
-			flt(self.gross_pay) * flt(self.exchange_rate), self.precision("base_gross_pay")
-		)
+		set_gross_pay_and_base_gross_pay()
 
 		if self.salary_structure:
 			self.calculate_component_amounts("deductions")
 
+			deduction_abbrs = [d.abbr for d in self.deductions]
+			self.update_dependent_components_recursively("earnings", deduction_abbrs)
+
+		set_gross_pay_and_base_gross_pay()
+		self.update_dependent_components_recursively("deductions", "gross_pay")
+		self.update_dependent_components_recursively("deductions", "base_gross_pay")
```

Comment:
Can this happen in one call? If any of them is found in the formula, update amount